common-service/src/main/java/org/sunbird/util/authentication/RequestInterceptor.java

```diff
@@ -0,0 +1,115 @@
+package org.sunbird.util.authentication;
+
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.request.HeaderParam;
+import play.mvc.Http;
+import play.mvc.Http.Request;
+
+/**
+ * This class will do the request header validation
+ *
+ * @author Amit Kumar
+ */
+public class RequestInterceptor {
+
+  private static List<String> restrictedUrlList;
+  private static List<String> publicUrlList;
+  private static List<String> privateUrlList;
+
+  public static List<String> getRestrictedUrlList() {
+    return restrictedUrlList;
+  }
+
+  public static void setRestrictedUrlList(List<String> restrictedUrlList) {
+    RequestInterceptor.restrictedUrlList = restrictedUrlList;
+  }
+
+  public static List<String> getPublicUrlList() {
+    return publicUrlList;
+  }
+
+  public static void setPublicUrlList(List<String> publicUrlList) {
+    RequestInterceptor.publicUrlList = publicUrlList;
+  }
+
+  public static List<String> getPrivateUrlList() {
+    return privateUrlList;
+  }
+
+  public static void setPrivateUrlList(List<String> privateUrlList) {
+    RequestInterceptor.privateUrlList = privateUrlList;
+  }
+
+  private RequestInterceptor() {}
+
+  /**
+   * This Method will do the request header validation.
```

Comment:
Authenticates given HTTP request context.

ctx HTTP play request context

User or Client ID for authenticated request. For unauthenticated requests, UNAUTHORIZED is returned.